server/src/com/google/refine/Refine.java

```diff
@@ -494,13 +495,32 @@ public void actionPerformed(ActionEvent e) {
     } 
     
     private void openBrowser() {
-        if (!Desktop.isDesktopSupported()) {
-            logger.warn("Java Desktop class not supported on this platform.  Please open %s in your browser",uri.toString());
+        if (!Desktop.getDesktop().isSupported(Desktop.Action.BROWSE)) {
+            try {
+                openBrowserFallback();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        } else {
+            try {
+                Desktop.getDesktop().browse(uri);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
         }
-        try {
-            Desktop.getDesktop().browse(uri);
-        } catch (IOException e) {
-            throw new RuntimeException(e);
+    }
+
+    private void openBrowserFallback() throws IOException {
+        Runtime rt = Runtime.getRuntime();
+
+        if(SystemUtils.IS_OS_WINDOWS) {
+            rt.exec("rundll32 url.dll,FileProtocolHandler " + uri);
+        } else if (SystemUtils.IS_OS_MAC_OSX) {
+            rt.exec("open " + uri);
+        } else if (SystemUtils.IS_OS_LINUX) {
+            rt.exec("xdg-open " + uri);
+        } else {
+            throw new RuntimeException("Unable to open browser");
```

Comment:
Perhaps in this last case we should resort to the previous behaviour (`logger.warn("Java Desktop class not supported on this platform.  Please open %s in your browser",uri.toString());`) because otherwise by throwing the exception we will likely stop OpenRefine completely, and that is probably not the desired behaviour.

Reply:
Done